dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/SdkFeatureTest.kt

```diff
@@ -273,6 +279,37 @@ internal abstract class SdkFeatureTest<T : Any, C : Configuration.Feature, F : S
         verify(mockDataFlusher).flush(testedFeature.uploader)
     }
 
+    @Test
+    fun `𝕄 migrate batch files 𝕎 initialize()`(
+        @StringForgery message: String
+    ) {
+        // Given
+        val fileName = System.currentTimeMillis().toString()
+        val oldFilesDir = File(appContext.fakeFilesDir, "dd-${featureDirName()}-v1")
+        oldFilesDir.mkdirs()
+        val oldBatchFile = File(oldFilesDir, fileName)
+        val content = "{\"message\":\"$message\"}"
+        oldBatchFile.writeText(content)
+
+        // When
+        testedFeature.initialize(appContext.mockInstance, fakeConfigurationFeature)
+
+        // Then
+        val newFilesDir = File(appContext.fakeCacheDir, "dd-${featureDirName()}-v1")
+        val newBatchFile = File(newFilesDir, fileName)
+
+        val captor = argumentCaptor<Runnable>()
+        // we expect one operation for the ConsentAware mechanism, and two for this migration
+        verify(coreFeature.mockPersistenceExecutor, times(3)).submit(captor.capture())
```

Comment:
It seems that the first operation (`ConsentAware` mechanism) is triggered at the `ConsentAwareFileOrchestrator` construction time, [here](https://github.com/DataDog/dd-sdk-android/blob/f6570f3b481d070669b891e2c76fcc96c8433cfc/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/advanced/ConsentAwareFileOrchestrator.kt#L26).

According to the test setup it will be `null` to `TrackingConsent.PENDING` migration which will [wipe](https://github.com/DataDog/dd-sdk-android/blob/f6570f3b481d070669b891e2c76fcc96c8433cfc/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/advanced/ConsentAwareFileMigrator.kt#L30) the data.

So that is leading me to the following question: why `null` to `PENDING` operation is wipe, but `PENDING` to `PENDING` is no-op, considering that `null` and `PENDING` is effectively the same - they share the same `pendingOrchestrator` according to [this](https://github.com/DataDog/dd-sdk-android/blob/f6570f3b481d070669b891e2c76fcc96c8433cfc/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/advanced/ConsentAwareFileOrchestrator.kt#L85).

Also it seems that we don't handle `filesDir` to `cacheDir` migration in case if app had `PENDING` state, did app upgrade and started again - in this case pending files in the `filesDir` will be lost and not moved.

Reply:
I think this is a rule we agreed upon long time ago that if the app is restarted and if the data was pending and there was no TrackingConsent value set yet, we will wipe out the data. In case the consent was already set to `PENDING` we keep the data for the moment when we will have the GO.

Reply:
Yes, indeed this is a rule that was defined. `null` to `Pending` means that the previous session was never granted, and since the app was closed in-between, it means we can't get any consent for that previous one so the data needs to be discarded.